sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/BloomFilterMightContain.scala

```diff
@@ -59,12 +61,25 @@ case class BloomFilterMightContain(
             if !subquery.containsPattern(OUTER_REFERENCE) =>
             TypeCheckResult.TypeCheckSuccess
           case _ =>
-            TypeCheckResult.TypeCheckFailure(s"The Bloom filter binary input to $prettyName " +
-              "should be either a constant value or a scalar subquery expression")
+            DataTypeMismatch(
+              errorSubClass = "BLOOM_FILTER_BINARY_OP_WRONG_TYPE",
+              messageParameters = Map(
+                "functionName" -> toSQLId(prettyName),
+                "actual" -> toSQLType(bloomFilterExpression.dataType)
```

Comment:
Type itself is not a problem, correct? More useful would be to show the non-constant or non-scalar subquery expr. How about:
`toSQLExpr(bloomFilterExpression)`

Reply:
> toSQLExpr(bloomFilterExpression)

Yes, I tested it and using `toSQLExpr` is more clearer and useful, it will return actual bloomFilterExpression.